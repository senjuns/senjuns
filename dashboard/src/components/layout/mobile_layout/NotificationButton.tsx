import React from 'react';
import styled from 'styled-components';
import notificationIcon from '../../../assets/svg/notification-button.svg';

interface NotificationButtonProps {}

const NotificationButton: React.FC<NotificationButtonProps> = () => {
  return (
    <Container>
      <img src={notificationIcon} />
      <NotificationNumber>{12}</NotificationNumber>
    </Container>
  );
};

const Container = styled.div`
  position: relative;
`;

const NotificationNumber = styled.div`
  position: absolute;
  font-size: 10px;
  font-weight: bold;
  color: white;
  background-color: #e86339;
  border-radius: 10px;
  right: calc(-1rem + 3px);
  top: -3px;
  padding: 2px;
`;

export default NotificationButton;
