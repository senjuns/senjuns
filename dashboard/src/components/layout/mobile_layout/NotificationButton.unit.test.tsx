// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import NotificationButton from './NotificationButton';

describe('<NotificationButton />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<NotificationButton />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
