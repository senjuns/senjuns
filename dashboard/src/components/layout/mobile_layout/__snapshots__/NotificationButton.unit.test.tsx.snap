// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationButton /> renders correctly. 1`] = `
<styled.div>
  <img
    src="notification-button.svg"
  />
  <styled.div>
    12
  </styled.div>
</styled.div>
`;
